tests/simplex/matrix_free_05.cc

```diff
@@ -0,0 +1,352 @@
+                                    degree);
+
+            for (unsigned int q = 0; q < fe_eval.n_q_points; ++q)
+              {
+                Tensor<1, dim, VectorizedArray<number>> average_value =
+                  fe_eval.get_value(q);
+                Tensor<1, dim, VectorizedArray<number>> average_valgrad =
+                  -fe_eval.get_normal_derivative(q);
+                average_valgrad += average_value * sigmaF;
+                fe_eval.submit_normal_derivative(-average_value, q);
+                fe_eval.submit_value(average_valgrad, q);
+              }
+
+            fe_eval.integrate_scatter(EvaluationFlags::values |
+                                        EvaluationFlags::gradients,
+                                      dst);
+          }
+      },
+      dst,
+      src,
+      true);
+  }
+
+private:
+  const MatrixFree<dim, double> &matrix_free;
+  const unsigned int             degree;
+};
+
+template <int dim>
+void
+test(const unsigned int degree)
+{
+  Triangulation<dim> tria;
+
+  unsigned int subdivisions = degree == 1 ? 16 : 8;
+
+
+  GridGenerator::subdivided_hyper_cube_with_simplices(tria, subdivisions);
+
+  FESystem<dim>      fe(FE_SimplexDGP<dim>(degree), dim);
+  QGaussSimplex<dim> quad(degree + 1);
+  MappingFE<dim>     mapping(FE_SimplexP<dim>(1));
+
+
+  DoFHandler<dim> dof_handler(tria);
+
+  dof_handler.distribute_dofs(fe);
+
+  AffineConstraints<double> constraints;
+  constraints.close();
+
+  const auto solve_and_postprocess =
+    [&](const auto &poisson_operator,
+        auto       &x,
+        auto       &b) -> std::pair<unsigned int, double> {
+    ReductionControl reduction_control(3000, 1e-7, 1e-2);
+    SolverCG<std::remove_reference_t<decltype(x)>> solver(reduction_control);
+
+    solver.solve(poisson_operator, x, b, PreconditionIdentity());
+
+    if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+      printf("Solved in %d iterations.\n", reduction_control.last_step());
+
+    constraints.distribute(x);
+
+#if 0
+    DataOut<dim> data_out;
+    data_out.attach_dof_handler(dof_handler);
+    x.update_ghost_values();
+    data_out.add_data_vector(dof_handler, x, "solution");
+    data_out.build_patches(mapping, degree);
+    data_out.write_vtu_with_pvtu_record("./",
+                                        "result-" + std::to_string(dim) + "-" +
+                                          std::to_string(degree) + "-" +
+                                          std::to_string(version),
+                                        0,
+                                        MPI_COMM_WORLD);
+#endif
+
+    Vector<double> difference(tria.n_active_cells());
+
+    deallog << "dim=" << dim << ' ';
+    deallog << "degree=" << degree << ' ';
+
+    VectorTools::integrate_difference(mapping,
+                                      dof_handler,
+                                      x,
+                                      Functions::ZeroFunction<dim>(),
```

Comment:
You need to select `dim` components here, the tester is complaining:
```suggestion
                                      Functions::ZeroFunction<dim>(dim),
```